vscode-extension/src/extension.ts

```diff
@@ -102,10 +103,29 @@ export async function activate(context: ExtensionContext) {
         });
       }
     );
+    const downloadDebugPayloadCommand = commands.registerCommand(
+      'metals.downloadDebugPayload',
+      async () => {
+        const byteArray = await client.sendRequest(ExecuteCommandRequest.type, {
+          command: "downloadDebugPayload",
+          arguments: [window.activeTextEditor.document.uri.toString()]
+        })
+        const workspaceRoot = workspace.workspaceFolders[0].uri.fsPath
+        const fileName = `${workspaceRoot}/.metals/debug-${Date.now()}.zip`
+        writeFile(fileName, new Buffer(byteArray), (err: Error) => {
```

Comment:
I suggest that we move as much as possible of the command implementation to the server side, so that clients only need to invoke the command and provide any optional UI. Can we write the zip from the server?

Reply:
I think you're right, it's much easier for the client.

I was still working under the assumption that server and client could technically live on different machines, but I think it's a bit of a stretch as of today.

Reply:
I agree with @laughedelic this should be handled fully server side. Remote LSP is an interesting application that is not supported currently and would require custom development to work.